src/lib/RichText/Converter/ListLineBreaks.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\FieldTypeRichText\RichText\Converter;
+
+use EzSystems\EzPlatformRichText\eZ\RichText\Converter;
+use DOMDocument;
+use DOMXPath;
+
+/**
+ * Class ListLineBreaks.
+ *
+ * Processes lists below <literallayout> tag.
+ */
+class ListLineBreaks implements Converter
+{
+    /**
+     * For all <itemizedList> and <orderedList> below <literallayout>, move the list after the <literallayout>, so that it is not inside it.
+     *
+     * @param \DOMDocument $document
+     *
+     * @return \DOMDocument
+     */
+    public function convert(DOMDocument $document)
+    {
+        $xpath = new DOMXPath($document);
+        $xpathExpression = '//ns:literallayout [descendant::ns:orderedlist|descendant::ns:itemizedlist]';
+        $ns = $document->documentElement ? $document->documentElement->namespaceURI ?: '' : '';
+        $xpath->registerNamespace('ns', $ns);
+        $elements = $xpath->query($xpathExpression) ?: [];
+
+        // elements are list of <literallayout> elements
+        foreach ($elements as $element) {
+            $listItemToMove = [];
+            $listItemNo = 0;
+            /** @var \DOMElement $element */
+            foreach ($element->childNodes as $childNode) {
+                /** @var \DOMNode $childNode */
+                if ($childNode instanceof \DOMElement) {
+                    if ($childNode->tagName == 'orderedlist' || $childNode->tagName == 'itemizedlist') {
+                        $listItemToMove[$listItemNo] = $childNode;
```

Comment:
Because you cannot change a DOMNodeList while you are looping over it.
[This comment](https://www.php.net/manual/en/class.domnodelist.php#83180) is old, but is still valid it seems

Reply:
> Because you cannot change a DOMNodeList while you are looping over it. [This comment](https://www.php.net/manual/en/class.domnodelist.php#83180) is old, but is still valid it seems

@vidarl if you see the full snippet I proposed at the end of the last review, it does't seem to make a difference if move node while looping or on the fly. So the question is - do we have some missing test coverage or is the comment actually not valid anymore? Could you provide an example input and output DOM for which the code would fail?

Reply:
While working on this, or maybe some other PR I experienced that messing with a DOMNodeList while looping over it caused problems. From that point I have just been very careful about the mater and created a second loop just to be safe. I tested your code now and it works fine so I think we are good then.